runtime/codert_vm/cnathelp.cpp

```diff
@@ -885,13 +885,69 @@ old_fast_jitPutFlattenableStaticField(J9VMThread *currentThread)
 void* J9FASTCALL
 old_slow_jitLoadFlattenableArrayElement(J9VMThread *currentThread)
 {
-	return NULL;
+	SLOW_JIT_HELPER_PROLOGUE();
+	j9object_t arrayObject = (j9object_t)currentThread->floatTemp1;
+	U_32 index = *((U_32 *)currentThread->floatTemp2);
+	void *addr = NULL;
+	if (NULL == arrayObject) {
+		buildJITResolveFrameForRuntimeHelper(currentThread, parmCount);
+		addr = setCurrentExceptionFromJIT(currentThread, J9VMCONSTANTPOOL_JAVALANGNULLPOINTEREXCEPTION, NULL);
+	} else {
+		j9object_t value = NULL;
+		U_32 arrayLength = J9INDEXABLEOBJECT_SIZE(_currentThread, arrayObject);
+		if (index >= arrayLength) {
+			currentThread->tempSlot = (UDATA)index;
+			buildJITResolveFrameForRuntimeHelper(currentThread, parmCount);
+			addr = setCurrentExceptionFromJIT(currentThread, J9VMCONSTANTPOOL_JAVALANGARRAYINDEXOUTOFBOUNDSEXCEPTION, NULL);
```

Comment:
This line will then go to JIT_RETURN_UDATA(value), returning NULL. I want to confirm that this is fine?

Reply:
if you are throwing an exception you dont need to set the return val, you can just jum straight to done

Reply:
or conversely, just move up the `JIT_RETURN_UDATA(value);` to after the NULL check for the return value